pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -478,9 +475,10 @@ private void stopHelixController() {
 
   private void stopPinotController() {
     try {
-      // Stopping ControllerLeadershipManager has to be done before stopping HelixResourceManager.
-      LOGGER.info("Stopping controller leadership manager");
-      _controllerLeadershipManager.stop();
+      // Stopping periodic tasks has to be done before stopping HelixResourceManager.
+      // Stop controller periodic task.
+      LOGGER.info("Stopping controller periodic tasks");
```

Comment:
We will need to stop lead controller manager so that we stop listening and reacting to state changes on leader controller resource?

Reply:
No, stopping controller will mark all the periodic tasks to stop. Each periodic task will firstly check whether to stop/run, and then check whether to run the task for a given table.